cumulus_library/studies/discovery/discovery_templates/code_system_pairs.sql.jinja

```diff
@@ -0,0 +1,44 @@
+{%- import 'syntax.sql.jinja' as syntax -%}
+
+{%- macro get_source_col(column_hierarchy) -%}
+{%- if column_hierarchy|length >1 or column_hierarchy[0][1].__name__ == 'list'-%}
+table_{{ column_hierarchy|length  }}.col_{{ column_hierarchy|length  }}
+{%- else -%}
+{{ column_hierarchy[0][0] }}
+{%-endif %}
+{%- endmacro -%}
+
+CREATE TABLE {{ output_table_name }} AS
+{%- for table in code_system_tables %}
+{%- if table.has_data %}
+SELECT DISTINCT
+    '{{ table.table_name }}' AS table_name,
+    '{{ table.column_display_name }}' AS column_name,
+    {{ get_source_col(table.column_hierarchy) }}.code,
+    {{ get_source_col(table.column_hierarchy) }}.display,
+    {{ get_source_col(table.column_hierarchy) }}.system
+FROM {{ table.table_name }}
+{%- if table.column_hierarchy|length >1 or table.column_hierarchy[0][1].__name__ == 'list' -%},
+UNNEST({{ table.column_hierarchy[0][0] }}) AS table_1 (col_1)
+{%- endif %}
```

Comment:
Maybe my brain is mush, but why does that first length check exist - you can't unnest a (non-list) struct field, right?

Reply:
so yeah - if you had a root level bare coding,  you wouldn't/couldn't do any unnesting